homeassistant/helpers/condition.py

```diff
@@ -234,24 +234,24 @@ def if_state(hass, variables=None):
 
 def sun(hass, before=None, after=None, before_offset=None, after_offset=None):
     """Test if current time matches sun requirements."""
-    now = dt_util.now().time()
+    now = dt_util.now()
+    utcnow = dt_util.as_utc(now)
     before_offset = before_offset or timedelta(0)
     after_offset = after_offset or timedelta(0)
+    entity = hass.data[sun_cmp.ENTITY_ID]
```

Comment:
This is not allowed.

All pieces of the system are set up in such a way that the only interaction between the parts is happening via the state machine. With this approach having a remote Hass instance would not be possible. Please revert.

The methods on top of the sun component are meant to extract the data from the state machine. If it's not in the state machine, it cannot be used.